File: samples/callbot.c
/* Minimal call/check poker bot — Botzone Traditional + LongRunning。
 * response=0 同时表示 call/check；首个响应后输出 LongRunning 标准握手。
 */
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    char *line = (char *)malloc(4000000);
    if (!line) return 1;
    int first_response = 1;
    while (fgets(line, 4000000, stdin)) {
        /* call/check 都是 0（Botzone 裸整数 0） */
        fputs("{\"response\":0}\n", stdout);
        if (first_response) {
            fputs(">>>BOTZONE_REQUEST_KEEP_RUNNING<<<\n", stdout);
            first_response = 0;
        }
        fflush(stdout);
    }
    free(line);
    return 0;
}

File: samples/callbot.py
#!/usr/bin/env python3
import json
import sys

KEEP_RUNNING = ">>>BOTZONE_REQUEST_KEEP_RUNNING<<<"


def current_request(envelope):
    if "request" in envelope:
        request = envelope["request"]
    else:
        requests = envelope.get("requests")
        if not isinstance(requests, list) or not requests:
            raise ValueError("missing requests")
        request = requests[-1]
    if not isinstance(request, dict):
        raise ValueError("request payload must be an object")
    return request


def main():
    first_response = True
    for line in sys.stdin:
        if not line.strip():
            continue

        envelope = json.loads(line)
        current_request(envelope)

        # Holdem: response=0 表示 call/check。
        print(json.dumps({"response": 0}, separators=(",", ":")), flush=True)

        if first_response:
            print(KEEP_RUNNING, flush=True)
            first_response = False


if __name__ == "__main__":
    main()
